src/App.tsx

```diff
@@ -1,26 +1,110 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import { getTodos, USER_ID } from './api/todos';
+import { Todo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
+import { ErrorComponent } from './components/ErrorComponent';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState<string | null>(null);
+  const [editingTodosId, setEditingTodosId] = useState<number[]>([]);
+  const [isActiveFilter, setIsActiveFilter] = useState(false);
+  const [filter, setFilter] = useState('all');
+
+  async function loadAllTodos() {
+    try {
+      const loadedTodos = await getTodos();
+
+      if (loadedTodos.length > 0) {
+        setIsActiveFilter(true);
+      }
+
+      setTodos(loadedTodos);
+      setFilteredTodos(loadedTodos);
+    } catch (e) {
+      setError('Unable to load todos');
+    }
+  }
+
+  useEffect(() => {
+    loadAllTodos();
+  }, []);
+
+  useEffect(() => {
+    setTimeout(() => {
+      setError(null);
+    }, 3000);
+  }, [error]);
+
+  useEffect(() => {
+    if (todos.length === 0) {
+      setIsActiveFilter(false);
+
+      return;
+    }
+
+    setIsActiveFilter(true);
+  }, [todos.length]);
+
+  useEffect(() => {
+    if (filter === 'completed') {
+      setFilteredTodos(prev => prev.filter(t => t.completed));
+    }
+
+    if (filter === 'active') {
+      setFilteredTodos(prev => prev.filter(t => !t.completed));
+    }
+  }, [filter, filteredTodos, todos]);
```

Comment:
It's not a good idea to have the `filteredTodos` as a dependency here. It causes your list to not render all the todos when you change the filter from `Active`/`Completed` to `All`.